micro-rdk/src/common/config_monitor.rs

```diff
@@ -0,0 +1,56 @@
+use super::app_client::{ AppClient, AppClientError, PeriodicAppClientTask};
+use futures_lite::Future;
+use std::pin::Pin;
+use std::time::Duration;
+use crate::proto::app::v1::ConfigResponse;
+
+pub struct ConfigMonitor<'a> {
+    restart_hook: Option<Box<dyn FnOnce() + 'a>>,
+    curr_config: ConfigResponse, //config for robot gotten from last robot startup, aka inputted from entry
+}
+
+impl<'a> ConfigMonitor<'a> {
+    pub fn new(restart_hook: impl FnOnce() + 'a, curr_config: ConfigResponse) -> Self {
+        Self {
+            restart_hook: Some(Box::new(restart_hook)),
+            curr_config: curr_config,
+        }
+    }
+
+    fn restart(&mut self) -> ! {
+        log::warn!("Config change detected - restarting or terminating now...");
+        (self.restart_hook.take().unwrap())();
+        unreachable!();
+    }
+}
+
+impl<'a> PeriodicAppClientTask for ConfigMonitor<'a> {
+    fn name(&self) -> &str {
+        "ConfigMonitor"
+    }
+
+    fn get_default_period(&self) -> Duration {
+        Duration::from_secs(10)
+    }
+
+    fn invoke<'c, 'b: 'c>(
+        &'b mut self,
+        app_client: &'c AppClient,
+    ) -> Pin<Box<dyn Future<Output = Result<Option<Duration>, AppClientError>> + 'c>> {
+        Box::pin(async move {
+            let app_config = get_app_config(app_client).await.unwrap();
+            match self.curr_config == app_config{
+                true => self.restart(),
+                false => Ok(Some(self.get_default_period())),
+            }
+        })
+    }
+
+}
+
+async fn get_app_config(client : &AppClient) -> Result<ConfigResponse, AppClientError>{
+    let (new_config, _cfg_received_datetime) = 
+            client.get_config(None).await.unwrap();
```

Comment:
An interesting case to consider is what currently does (and what should) happen if the configuration becomes invalid (e.g. the robot was deleted), such that you can no longer obtain it with `get_config`.

Reply:
Right now, I believe it core dumps and has a error within the monitor, but if the config is invalid we could just grab it from the config cache (in the next ticket as discussed above)?

Reply:
do we store a default empty config in config cache in case the config is invalid on the very first config initial setup?

Reply:
No. I filed https://viam.atlassian.net/browse/RSDK-8127 to track this general area.